web/src/redux/persistent/projects/entry-points/reducer.ts

```diff
@@ -18,9 +22,22 @@ import { isCrud, crudReducer } from '../../crudRedux'
 // state is an object where the keys are the entry headerHashes of "EntryPoints"
 // and the values are modified versions of the EntryPoint data structures that
 // also contain their headerHash on those objects
-const defaultState = {}
+type State = {
+  [cellId: CellIdString]: {
+    [headerHash: HeaderHashB64]: {
+      color: String,
+      creatorAddress: AgentPubKeyB64,
+      createdAt: number, //f64,
+      outcomeAddress: HeaderHashB64,
+      isImported: boolean,
+      // field additional to EntryPoint
```

Comment:
Same here. We could say there is a type called WithHeaderHash that has headerHash on it, and all these types that need that can intersect with it. 

That way we can avoid duplicate definition of each entry type in typescript